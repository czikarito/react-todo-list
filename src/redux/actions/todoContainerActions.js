import { ADD_TODO, COMPLETE_TODO } from "../actionTypes";

export const addTodo = value => {
  return {
    type: ADD_TODO,
    newTodo: {
      id: Math.random(),
      title: value,
      completed: false
    }
  }
}

export const completeTodo = (id) => {
  return {
    type: COMPLETE_TODO,
    id: id
  }
};
