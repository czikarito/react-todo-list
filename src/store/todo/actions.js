import { ADD_TODO, COMPLETE_TODO, EDIT_TODO } from './actionTypes';

export const addTodo = payload => {
  return {
    type: ADD_TODO,
    payload
  };
};

export const completeTodo = id => {
  return {
    type: COMPLETE_TODO,
    payload: { id }
  };
};

export const editTodo = payload => {
  return {
    type: EDIT_TODO,
    payload
  };
};
