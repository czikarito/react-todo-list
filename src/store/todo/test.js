import reducer from '.';
import * as actions from './actions';

describe('todo reducer', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should add todo', () => {
    expect(
      reducer(undefined, {
        type: 'ADD_TODO',
        payload: {
          title: '4',
          author: 'Pawel'
        }
      }).todos.length
    ).toEqual(4);
  });

  it('should complete todo', () => {
    expect(
      reducer(undefined, {
        type: 'COMPLETE_TODO',
        payload: {
          id: 1
        }
      }).todos[0].completed
    ).toBe(true);
  });

  it('should edit todo', () => {
    const editedTodo = {
      id: 1,
      title: 'New title',
      author: 'test',
      completed: false
    };

    expect(
      reducer(undefined, {
        type: 'EDIT_TODO',
        payload: editedTodo
      }).todos[0]
    ).toEqual(editedTodo);
  });
});
