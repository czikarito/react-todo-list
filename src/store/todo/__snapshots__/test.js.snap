exports[`todo reducer should return initial state 1`] = `
Object {
  "todos": Array [
    Object {
      "author": "Pawel",
      "completed": false,
      "id": 1,
      "title": "Clean a room",
    },
    Object {
      "author": "Karina",
      "completed": false,
      "id": 2,
      "title": "Buy a milk",
    },
    Object {
      "author": "Pawel",
      "completed": false,
      "id": 3,
      "title": "Go out with the dog",
    },
  ],
}
`;
